bitcoin/src/psbt/mod.rs

```diff
@@ -351,14 +351,14 @@ impl PartiallySignedTransaction {
                 )
                 .ok_or(SignError::NotWpkh)?;
                 let sighash =
-                    cache.segwit_signature_hash(input_index, &script_code, utxo.value, hash_ty)?;
+                    cache.segwit_signature_hash(input_index, &script_code, utxo.value.to_sat(), hash_ty)?;
                 Ok((Message::from(sighash), hash_ty))
             }
             Wsh | ShWsh => {
                 let script_code =
                     input.witness_script.as_ref().ok_or(SignError::MissingWitnessScript)?;
                 let sighash =
-                    cache.segwit_signature_hash(input_index, script_code, utxo.value, hash_ty)?;
+                    cache.segwit_signature_hash(input_index, script_code, utxo.value.to_sat(), hash_ty)?;
```

Comment:
Looks like changing the `segwit_signature_hash` method to accept `Amount` instead would be better.